engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -147,9 +147,65 @@
     /**
      * Accumulated delays due to intracycle sleeps for the current cycle (or previous, if idle).
      */
-
     private long currentCycleSleepTotalNanos = 0L;
 
+    public static class AccumulatedCycleStats {
+        /**
+         * Total cycles run.
+         */
+        public long totalCycles = 0L;
+        /**
+         * Cycles run no exceeding their time budget.
+         */
+        public long totalCyclesOnBudget = 0L;
```

Comment:
I sort of figure why bother computing that here, when you can just do that afterwards when you are computing other things.

Reply:
The idea was that the target cycle time may change (but then I changed it during debugging to the constant and forgot to revert that, also ref your comment below)